tests/dialects/test_seq.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+
+from xdsl.dialects.arith import Constant
+from xdsl.dialects.builtin import (
+    IntegerAttr,
+    i32,
+)
+from xdsl.dialects.seq import (
+    ClockDivider,
+    ClockType,
+)
+from xdsl.utils.exceptions import VerifyException
+
+
+def test_clockdivider_verify():
+    clock_div = ClockDivider.build(
+        operands=[Constant(ClockType(), ClockType())],
+        attributes={
+            "pow2": IntegerAttr(512, i32),
+        },
+        result_types=[ClockType()],
+    )
```

Comment:
This is great.
I'd simplify by not using the builder methods and using the `test` dialect for whatever is not strictly needed for testing.

```suggestion
    clock_div = ClockDivider(
        test.TestOp.create(result_types=[clock]), IntegerAttr(513, i32)
    )
```

Reply:
nit:
```suggestion
    clock_div = ClockDivider(
        test.TestSSAValue(clock), IntegerAttr(513, i32)
    )
```